api/src/wfl/service/datarepo.clj

```diff
@@ -117,3 +118,126 @@
       util/response-body-json
       :id
       poll-job))
+
+(defn create-snapshot
+  "Create a snapshot from standard SQL query,
+   assert or row-ids, based on `snapshot-request`.
+
+   See `SnapshotRequestModel` in the
+   DataRepo swagger page for more information.
+   https://jade.datarepo-dev.broadinstitute.org/swagger-ui.html#/
+
+   Note the TDR is under active development,
+   the endpoint spec is getting changed so the
+   spec in this function is not consistent with
+   the TDR Swagger page in order to make the
+   request work."
+  [snapshot-request]
+  (-> (repository "snapshots")
+      (http/post {:headers      (auth/get-service-account-header)
+                  :content-type :application/json
+                  :form-params  snapshot-request})
+      util/response-body-json
+      :id
+      poll-job))
+
+(defn list-snapshots
+  "List all Snapshots the caller has access to.
+   Hard-coded to return 999 pages for now.
+
+   Parameters
+   ----------
+   dataset-ids      - Optionally filter the result snapshots
+                      where provided datasets are source datasets.
+
+   Example
+   -------
+     (list-snapshots)
+     (list-snapshots \"48a51f71-6bab-483d-a270-3f9ebfb241cd\" \"85efdfea-52fb-4698-bee6-eef76104a7f4\")"
+  [& dataset-ids]
+  (letfn [(maybe [m k v] (if (seq v) (assoc m k {:datasetIds v}) m))]
+    (-> (http/get (repository "snapshots")
+                  (-> {:headers (auth/get-service-account-header)
+                       :query-params {:limit 999}}
+                      (maybe :query-params dataset-ids)))
```

Comment:
Again, I don't think the threading helps here, it's just adding parenthesis (in a lisp dialect? Heaven forfend!). Perhaps?
```clojure
(maybe
  {:headers      (auth/get-service-account-header)
   :query-params {:limit 999}}
  :query-params dataset-ids)
```

Reply:
Totally makes sense! I'm just a bit obsessed with the beautiful threadings recently 🤓